src/main/webapp/app/exercises/programming/hestia/git-diff-report/git-diff-file-panel.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, Input, OnInit, ViewEncapsulation } from '@angular/core';
+import { ProgrammingExerciseGitDiffEntry } from 'app/entities/hestia/programming-exercise-git-diff-entry.model';
+import { faAngleDown, faAngleUp } from '@fortawesome/free-solid-svg-icons';
+
+@Component({
+    selector: 'jhi-git-diff-file-panel',
+    templateUrl: './git-diff-file-panel.component.html',
+    styleUrls: ['./git-diff-file-panel.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+})
+export class GitDiffFilePanelComponent implements OnInit {
+    @Input()
+    diffEntries: ProgrammingExerciseGitDiffEntry[];
+
+    @Input()
+    templateFileContent: string | undefined;
+
+    @Input()
+    solutionFileContent: string | undefined;
+
+    @Input()
+    numberOfContextLines = 3;
+
+    previousFilePath: string | undefined;
+    filePath: string | undefined;
+
+    addedLineCount: number;
+    removedLineCount: number;
+
+    faAngleUp = faAngleUp;
+    faAngleDown = faAngleDown;
+
+    ngOnInit(): void {
+        this.filePath = this.diffEntries
+            .map((entry) => entry.filePath)
+            .filter((filePath) => filePath)
+            .first();
+
+        this.previousFilePath = this.diffEntries
+            .map((entry) => entry.previousFilePath)
+            .filter((filePath) => filePath)
+            .first();
+
+        this.addedLineCount = this.diffEntries
+            .flatMap((entry) => {
+                if (entry && entry.filePath && entry.startLine && entry.lineCount) {
```

Comment:
What happens here if the entry does not fulfill the if-conditions and would it make sense to move this condition to a `filter()` function?